ddterm/app/application.js

```diff
@@ -553,8 +555,11 @@ class Application extends Gtk.Application {
         });
 
         this.window.connect('destroy', source => {
-            if (source === this.window)
+            if (source === this.window) {
+                // All tabs are closed, remove the session file
+                GLib.unlink(this.session_file_path);
```

Comment:
Doesn't it remove the session file every time the window is closed? Including, for example, when you log out while ddterm is running (so the session gets saved and then immediately removed).

IMO the file should be removed in `save_session()` instead - when there's nothing to save, instead of writing an empty file.

Reply:
Or maybe forget about this case (no tabs/empty file) for now. Existing code writes an empty file, and it doesn't cause any problems. This can be a separate "optimization" PR.

Just don't remove the session file on `destroy`.

Reply:
Tested this branch again and, to my surprise, the session gets saved. Apparently, `destroy` isn't emitted when the session manager terminates the app. But when closing all tabs, an empty file gets written from `save_session()` first, then it's removed here - not the best solution.

Reply:
Oh, `.save_session()` actually doesn't get called when you close all tabs... You're right, `unlink()` is necessary when closing the window this way. But still, this is confusing - especially the fact that `destroy` signal isn't always emitted.

I propose the following solution:
- Unlink the file in `.save_session()`
- Add `.save_session()` call before `.close()`

Reply:
changed according to your proposal, at least I tried, not sure if my approach with the windows close signal is correct